Dockerfile

```diff
@@ -0,0 +1,30 @@
+# Copyright 2019 The OPA Authors.  All rights reserved.
+# Use of this source code is governed by an Apache2
+# license that can be found in the LICENSE file.
+
+# global ARG vars, see https://github.com/moby/moby/issues/37345
+ARG GOVERSION=1.12.9
+ARG GOARCH=amd64
+ARG GOOS=linux
+ARG VARIANT
+
+# multi-stage: build in golang container, copy into gcr.io/distroless/base
+FROM golang:${GOVERSION} AS build
+WORKDIR /go/src/github.com/open-policy-agent/opa
+COPY . .
+RUN make deps build
+
+FROM gcr.io/distroless/base${VARIANT}
+ARG GOARCH
+ARG GOOS
+ARG USER=0
+MAINTAINER Torin Sandall <torinsandall@gmail.com>
+COPY --from=build /go/src/github.com/open-policy-agent/opa/opa_${GOOS}_${GOARCH} /opa
+
+# Any non-zero number will do, and unfortunately a named user will not,
+# as k8s pod securityContext runAsNonRoot can't resolve the user ID:
+# https://github.com/kubernetes/kubernetes/issues/40958
+USER ${USER}
```

Comment:
I'm not familiar enough with Docker to know what happens when ${USER} is not defined. Can you put a comment in here?